cmd/kubeadm/app/componentconfigs/kubeproxy.go

```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/pkg/errors"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/klog/v2"
-	kubeproxyconfig "k8s.io/kube-proxy/config/v1alpha1"
```

Comment:
I'm not sure how to make kubeadm work with both v1alpha1 and v1alpha2 kube proxy config

Reply:
this change needs a discussion in a KEP.

the original linked KEP never answered the questions on how to migrate users from one version to another.
if we know the answers we can draft a plan.

right now this is just a PR that introduces some good features, but can break and/or confuse many users.

Reply:
a summary for kubeadm to support this:
- on kubeadm init  / cluster creation v1alpha2 must be used and stored in the config map
- kubeadm defaults must be patched everywhere to use the new API version
- on kubeadm upgrade ,the kube-proxy config map stored in the cluster must be mutated to v1alpha2

proposal:

ideally kube-proxy must export a CLI option such as `kube-proxy --convert-config-input --convert-config-output` that can take a v1alpha1 and produce a v1alpha2, this is ALA `kubectl convert`.
only kube-proxy (the component binary) keeps track of it's internal config and can perform this conversion.

the command can accept a set of API structs separated by `---` but only migrated the known kube-proxy API kinds.

this process must be documented somewhere at the k8s.io website.
i can comment this again on the KEP update PR, once it's open.

Reply:
that's great. good to know.
it just has to be documented at the right places at the k/website.

Reply:
https://github.com/kubernetes/enhancements/pull/4337